content/learn/book/migration-guides/0.9-0.10/_index.md

```diff
@@ -0,0 +1,1043 @@
+- `with_run_criteria` has been renamed to `run_if`. Run criteria have been renamed to run conditions for clarity, and should now simply return a bool.
+- States have been dramatically simplified: there is no longer a “state stack”. To queue a transition to the next state, call `NextState::set`
+- Strings can no longer be used as a `SystemLabel` or `SystemSet`. Use a type, or use the system function instead.
+
+### [Pipelined Rendering](https://github.com/bevyengine/bevy/pull/6503)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Rendering</div>
+    <div class="migration-guide-area-tag">Tasks</div>
+</div>
+
+__App `runner` and SubApp `extract` functions are now required to be Send__
+
+This was changed to enable pipelined rendering. If this breaks your use case please report it as these new bounds might be able to be relaxed.
+
+### [Remove ImageMode](https://github.com/bevyengine/bevy/pull/6674)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Rendering</div>
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+`ImageNode` never worked, if you were using it please create an issue.
+
+### [Rename the `background_color` of 'ExtractedUiNode` to `color`](https://github.com/bevyengine/bevy/pull/7452)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Rendering</div>
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+- The `background_color` field of `ExtractedUiNode` is now named `color`.
+
+### [Make spawn_dynamic return InstanceId](https://github.com/bevyengine/bevy/pull/6663)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Scenes</div>
+</div>
+
+<!-- TODO no migration required, will remove it later-->
+
+### [Parallelized transform propagation](https://github.com/bevyengine/bevy/pull/4775)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Transform</div>
+</div>
+
+<!-- TODO no migration required, will remove it later-->
+
+### [Remove the `GlobalTransform::translation_mut` method](https://github.com/bevyengine/bevy/pull/7134)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">Transform</div>
+    <div class="migration-guide-area-tag">Hierarchy</div>
+</div>
+
+`GlobalTransform::translation_mut` has been removed without alternative,
+if you were relying on this, update the `Transform` instead. If the given entity
+had children or parent, you may need to remove its parent to make its transform
+independent (in which case the new `Commands::set_parent_in_place` and
+`Commands::remove_parent_in_place` may be of interest)
+
+Bevy may add in the future a way to toggle transform propagation on
+an entity basis.
+
+### [Flip UI image](https://github.com/bevyengine/bevy/pull/6292)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">UI</div>
+</div>
+
+<!-- TODO -->

```

Comment:
```suggestion
- `UiImage` is a struct now, so use `UiImage::new(handler)` instead of `UiImage(handler)`
- `UiImage` no longer implements `Deref` and `DereftMut`, so use `&image.texture` or ``&mut image.texture` instead
```
AFAICT this one needs a migration guide, since `UiImage` is now a struct instead of a newtype, but the PR has none. This migration guide was my best guess by reading the PR.